pkg/instance/instance.go

```diff
@@ -0,0 +1,588 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/longhorn/longhorn-instance-manager/pkg/client"
+	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
+	"github.com/longhorn/longhorn-instance-manager/pkg/meta"
+	"github.com/longhorn/longhorn-instance-manager/pkg/types"
+)
+
+type Server struct {
```

Comment:
Cache `ProcessManagerClient` here.
BTW, does `ProcessManagerClient` need to care about TLS now?

Reply:
Do you mean we don't care about TLS for ProcessManagerClient because it is an internal service?

Reply:
Yes.

Reply:
Yeah, we still need TLS for ProcessManagerClient because of the backward compatibility.

Reply:
So,
The client codes still need TLS because of the communication to old IMs. TLS support for the process manager server can be removed. Right?